File: tests/Automation/demo/src/main/java/pageObjects/Login_Page.java
package pageObjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class Login_Page {

//    pass driver to this class
    public Login_Page(WebDriver driver)
    {
        PageFactory.initElements(driver,this);
    }
//    ************************************************************************************
//      input = i_
//      button = b_

//    ************************************************************************************
    @FindBy(xpath = "//input[@id='Input_Email']")
    public WebElement i_Email;

    @FindBy(xpath = "//input[@id='Input_Password']")
    public WebElement i_Password;

    @FindBy(xpath = "//button[text()='Log in']")
    public WebElement b_Login;
}

File: tests/eShopOnWebE2ETests/PageObjects/LoginPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;

namespace eShopOnWebE2ETests.PageObjects
{
    public class LoginPage
    {
        private IWebDriver _driver;

        public LoginPage(IWebDriver driver)
        {
            _driver = driver;
        }
        private IWebElement LoginButton => _driver.FindElement(By.XPath("//a[contains(text(),'Login')]"));
        private IWebElement UserNameInput => _driver.FindElement(By.XPath("//input[@id='Input_Email']"));
        private IWebElement PasswordInput => _driver.FindElement(By.XPath("//input[@id='Input_Password']"));
        private IWebElement SubmitLoginButton => _driver.FindElement(By.XPath("//button[normalize-space()='Log in']"));
        private IWebElement LoginPageTitle => _driver.FindElement(By.XPath("//h2[normalize-space()='Log in']"));
        public void EnterUsername(string username)
        {
            UserNameInput.SendKeys(username);
        }
        public void EnterPassword(string password)
        {
            PasswordInput.SendKeys(password);
        }
        public void ClickLogin()
        {
            LoginButton.Click();
        }
        public string GetThePageTitle()
        {
            return LoginPageTitle.Text;
        }
        public void SubmitLogin()
        {
            SubmitLoginButton.Click();
        }
    }
}
